include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h

```diff
@@ -504,10 +505,10 @@ __radix_sort_reorder_submit(_ExecutionPolicy&& __exec, ::std::size_t __segments,
                 // find offsets for the same values within a segment and fill the resulting buffer
                 for (::std::size_t __val_idx = __seg_start + __self_lidx; __val_idx < __seg_end; __val_idx += __sg_size)
                 {
-                    _InputT __in_val = __input_rng[__val_idx];
+                    auto __in_val = __input_rng[__val_idx];
```

Comment:
Potential problem with that is we could move the object. If it's tuple, it might contain rvalues. Let's discuss if you have questions, I might be wrong.

Reply:
Let's discuss it offline